packages/chain-bridge/src/adapters/StargateAdapter.sol

```diff
@@ -952,239 +774,28 @@ contract StargateAdapter is
                 (address, address, uint256, uint256, bytes32, address)
             );
 
-        // Basic validation - fail fast
-        if (
-            receivedAsset == address(0) ||
-            recipient == address(0) ||
-            amount == 0
-        ) {
-            revert InvalidParams();
-        }
-
-        // Check adapter balance
-        uint256 adapterBalance = IERC20(receivedAsset).balanceOf(address(this));
-        if (adapterBalance < amount) {
-            revert InsufficientBalance();
-        }
-
-        // Try to deliver assets - if it fails, hold them for governance recovery
-        bool success = _tryDeliverAssets(
-            recipient,
-            receivedAsset,
-            amount,
+        // -----------------------------------------------------------------
+        // 1. Forward the tokens the adapter just received to the router
+        // -----------------------------------------------------------------
+        IERC20(receivedAsset).safeTransfer(bridgeRouter(), amount);
+
+        // -----------------------------------------------------------------
+        // 2. Build the payload the end-recipient expects
+        //    (same shape as before, just without the token/amount fields)
+        // -----------------------------------------------------------------
+        bytes memory payload = abi.encode(operationId, originator, sourceAsset);
```

Comment:
the structure is not documented - and it has to be decoded by the receipient - would be good to have all the common message types properly typed with struct. Same goes for : https://github.com/OasisDEX/summer-earn-protocol/blob/f32a43112c2a257ea1c76aa58c6caaf1b9e09818/packages/chain-bridge/src/adapters/StargateAdapter.sol#L765-L775